### PR TITLE
EPMRPP-42422 || Test case id impl

### PR DESCRIPTION
**Pull request by Yumfriez**

### CODE REVIEW

File: src/main/java/com/epam/reportportal/junit/ParallelRunningHandler.java [hunk 1]
```diff
@@ -377,6 +383,49 @@ protected FinishTestItemRQ buildFinishStepRq(FrameworkMethod method, String stat
...
+
+	protected Optional<Integer> getParameterizedTestCaseId(Object target, String codeRef) {
+
+		com.google.common.base.Optional<Map<String, Object>> params = ((ArtifactParams) target).getParameters();
+		if (params.isPresent()) {
+			return Arrays.stream(target.getClass().getDeclaredFields())
+					.filter(field -> params.get().containsKey(field.getName()) && field.getDeclaredAnnotation(TestCaseIdKey.class) != null)
+					.findFirst()
+					.map(testCaseIdField -> {
+						TestCaseIdKey testCaseIdKeyAnnotation = testCaseIdField.getDeclaredAnnotation(TestCaseIdKey.class);
```
Comment (pbortnik) [inline src/main/java/com/epam/reportportal/junit/ParallelRunningHandler.java]: move it to a separate method with a clear name

File: src/main/java/com/epam/reportportal/junit/ParallelRunningHandler.java [hunk 2]
```diff
@@ -377,6 +383,49 @@ protected FinishTestItemRQ buildFinishStepRq(FrameworkMethod method, String stat
...
+				}
+			});
+
+		}
+		return Optional.of(Objects.hashCode(codeRef));
+	}
+
+	protected Optional<Integer> getParameterizedTestCaseId(Object target, String codeRef) {
+
+		com.google.common.base.Optional<Map<String, Object>> params = ((ArtifactParams) target).getParameters();
```
Comment (pbortnik) [inline src/main/java/com/epam/reportportal/junit/ParallelRunningHandler.java]: maybe java.util.Optional?

Comment (Yumfriez) [reply src/main/java/com/epam/reportportal/junit/ParallelRunningHandler.java]: JUnit ArtifactParams returns guava's Optional, so it's not possible

File: src/main/java/com/epam/reportportal/junit/ParallelRunningHandler.java (hunk 1 above)
Comment (Yumfriez) [reply src/main/java/com/epam/reportportal/junit/ParallelRunningHandler.java]: done

**Pull request merged**